src/mapeo-project.js

```diff
@@ -220,3 +279,29 @@ function extractEditableProjectSettings(projectDoc) {
   const { schemaName, ...result } = valueOf(projectDoc)
   return result
 }
+
+/**
+ * Return a map of namespace -> core keypair
+ *
+ * For the project owner the keypair for the 'auth' namespace is the projectKey
+ * and projectSecretKey. In all other cases keypairs are derived from the
+ * project key
+ *
+ * @param {object} opts
+ * @param {Buffer} opts.projectKey
+ * @param {Buffer} [opts.projectSecretKey]
+ * @param {import('@mapeo/crypto').KeyManager} opts.keyManager
+ */
+function getCoreKeypairs({ projectKey, projectSecretKey, keyManager }) {
+  /** @type {Record<string, import('./types.js').KeyPair>} */
+  const keypairs = {}
+  for (const namespace of NAMESPACES) {
+    keypairs[namespace] =
+      namespace === 'auth' && projectSecretKey
+        ? { publicKey: projectKey, secretKey: projectSecretKey }
+        : keyManager.getHypercoreKeypair(namespace, projectKey)
+  }
+  return /** @type {Record<import('./core-manager/core-index.js').Namespace, import('./types.js').KeyPair>} */ (
+    keypairs
+  )
+}
```

Comment:
not sure if any better from a type perspective but for any function that does casting internally, I typically prefer to add an an explicit `returns` annotation in the function definition:
 
```suggestion
/**
 * Return a map of namespace -> core keypair
 *
 * For the project owner the keypair for the 'auth' namespace is the projectKey
 * and projectSecretKey. In all other cases keypairs are derived from the
 * project key
 *
 * @param {object} opts
 * @param {Buffer} opts.projectKey
 * @param {Buffer} [opts.projectSecretKey]
 * @param {import('@mapeo/crypto').KeyManager} opts.keyManager
 * @returns {Record<import('./core-manager/core-index.js').Namespace, import('./types.js').KeyPair>}
 */
function getCoreKeypairs({ projectKey, projectSecretKey, keyManager }) {
  const keypairs =
    /** @type {Record<import('./core-manager/core-index.js').Namespace, import('./types.js').KeyPair>} */ ({})

  for (const namespace of NAMESPACES) {
    keypairs[namespace] =
      namespace === 'auth' && projectSecretKey
        ? { publicKey: projectKey, secretKey: projectSecretKey }
        : keyManager.getHypercoreKeypair(namespace, projectKey)
  }

  return keypairs
}
```

Suggestion diff isn't clear but basically:

- add return type in JSDoc definition
- add type cast on `keyPairs` variable declaration
- remove type cast from return value